flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/ProcessTableFunctionTestPrograms.java

```diff
@@ -0,0 +1,266 @@
+                                    .addSchema(BASE_SINK_SCHEMA)
+                                    .consumedValues("+I[{42, true}, 1]", "+I[{42, true}, 2]")
+                                    .build())
+                    .runSql(
+                            "INSERT INTO sink SELECT * FROM f(i => 42, b => CAST('TRUE' AS BOOLEAN))")
+                    .build();
+
+    public static final TableTestProgram PROCESS_TABLE_AS_ROW =
+            TableTestProgram.of("process-row", "table with row semantics")
+                    .setupTemporarySystemFunction("f", TableAsRowFunction.class)
+                    .setupSql(BASIC_VALUES)
+                    .setupTableSink(
+                            SinkTestStep.newBuilder("sink")
+                                    .addSchema(BASE_SINK_SCHEMA)
+                                    .consumedValues(
+                                            "+I[{+I[Bob, 12], 1}, 1]",
+                                            "+I[{+I[Bob, 12], 1}, 2]",
+                                            "+I[{+I[Alice, 42], 1}, 1]",
+                                            "+I[{+I[Alice, 42], 1}, 2]")
+                                    .build())
+                    .runSql("INSERT INTO sink SELECT * FROM f(r => TABLE t, i => 1)")
+                    .build();
+
+    public static final TableTestProgram PROCESS_TYPED_TABLE_AS_ROW =
+            TableTestProgram.of("process-typed-row", "typed table with row semantics")
+                    .setupTemporarySystemFunction("f", TypedTableAsRowFunction.class)
+                    .setupSql(BASIC_VALUES)
+                    .setupTableSink(
+                            SinkTestStep.newBuilder("sink")
+                                    .addSchema(BASE_SINK_SCHEMA)
+                                    .consumedValues(
+                                            "+I[{User(s='Bob', i=12), 1}, 1]",
+                                            "+I[{User(s='Bob', i=12), 1}, 2]",
+                                            "+I[{User(s='Alice', i=42), 1}, 1]",
+                                            "+I[{User(s='Alice', i=42), 1}, 2]")
+                                    .build())
+                    .runSql("INSERT INTO sink SELECT * FROM f(u => TABLE t, i => 1)")
+                    .build();
+
+    public static final TableTestProgram PROCESS_TABLE_AS_SET =
+            TableTestProgram.of("process-set", "table with set semantics")
+                    .setupTemporarySystemFunction("f", TableAsSetFunction.class)
+                    .setupSql(BASIC_VALUES)
+                    .setupTableSink(
+                            SinkTestStep.newBuilder("sink")
+                                    .addSchema(KEYED_BASE_SINK_SCHEMA)
+                                    .consumedValues(
+                                            "+I[Bob, {+I[Bob, 12], 1}, 1]",
+                                            "+I[Bob, {+I[Bob, 12], 1}, 2]",
```

Comment:
I am not sure myself if we need to change that, but I was slightly confused why this was printed twice if it's grouped by the name. I understood when I checked the function implementation (that every record is emitted twice), but made me wondering at first.

Reply:
I'm also annoyed by it nowadays. I will remove this.